search/services/index/classic.py

```diff
@@ -0,0 +1,103 @@
+"""Translate classic API `Phrase` objects to Elasticsearch DSL."""
+
+from typing import Dict, Callable
+
+from elasticsearch_dsl import Q, Search
+from elasticsearch_dsl.query import QueryString
+
+from ...domain import ClassicAPIQuery, Phrase, Term, Field, Operator
+from .prepare import SEARCH_FIELDS
+
+
+def classic_search(search: Search, query: ClassicAPIQuery) -> Search:
+    """
+    Prepare a :class:`.Search` from a :class:`.ClassicAPIQuery`.
+
+    Parameters
+    ----------
+    search : :class:`.Search`
+        An Elasticsearch search in preparation.
+    query : :class:`.ClassicAPIQuery`
+        An query originating from the Classic API.
+
+    Returns
+    -------
+    :class:`.Search`
+        The passed ES search object, updated with specific query parameters
+        that implement the advanced query.
+
+    """
+    # Initialize query
+    if query.phrase:
+        dsl_query = _phrase_to_query(query.phrase)
+    else:
+        dsl_query = Q()
+
+    # filter id_list if necessary
+    if query.id_list:
```

Comment:
Same comments as on Slack just now. The problem is:

```python
>>> Q() | Q('match', foo='bar')
MatchAll()
```
otoh:
```python
>>> from functools import reduce
>>> from operator import ior
>>> reduce(ior, [Q('match', foo='bar'), Q('term', baz='bat')])
Bool(should=[Match(foo='bar'), Term(baz='bat')])
```
but also see:
https://www.elastic.co/guide/en/elasticsearch/reference/6.4/query-dsl-terms-query.html
e.g.
```python
>>> Q('terms', foo=['bar', 'baz', 'bat'])
Terms(foo=['bar', 'baz', 'bat'])
```